src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/CreateCitizenClaimCallbackHandler.java

```diff
@@ -0,0 +1,145 @@
+        UserService userService
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.issueDateCalculator = issueDateCalculator;
+        this.referenceNumberRepository = referenceNumberRepository;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.caseMapper = caseMapper;
+        this.paymentsService = paymentsService;
+        this.eventProducer = eventProducer;
+        this.userService = userService;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return callbacks;
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse createCitizenClaim(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        logger.info("Created citizen case for callback of type {}, claim with external id {}",
+            callbackParams.getType(),
+            claim.getExternalId());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+        Payment payment = paymentsService.retrievePayment(authorisation, claim);
+
+        if (payment.getStatus() != PaymentStatus.SUCCESS) {
+            logger.info("Payment not successful for claim with external id {}", claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Payment not successful"))
+                .build();
+        }
+
+        LocalDate issuedOn = issueDateCalculator.calculateIssueDay(nowInLocalZone());
+
+        Claim updatedClaim = claim.toBuilder()
+            .channel(ChannelType.CITIZEN)
+            .claimData(claim.getClaimData().toBuilder()
+                .payment(payment)
+                .build())
+            .referenceNumber(referenceNumberRepository.getReferenceNumberForCitizen())
+            .issuedOn(issuedOn)
+            .responseDeadline(responseDeadlineCalculator.calculateResponseDeadline(issuedOn))
+            .build();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetailsConverter.convertToMap(caseMapper.to(updatedClaim)))
+            .build();
+    }
+
+    private CallbackResponse startClaimIssuedPostOperations(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        logger.info("Created citizen case for callback of type {}, claim with external id {}",
+            callbackParams.getType(),
+            claim.getExternalId());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+        User user = userService.getUser(authorisation);
+        eventProducer.createClaimCreatedEvent(
```

Comment:
As we are adding this here, we need to remove the same call from `ClaimService` otherwise post operations will be invoked twice.

Reply:
Nope it won't as we are invoking `saveCitizenClaim` and not `saveClaim`

Reply:
As discussed on Slack - seems the `ClaimService`'s invocation won't happen in an IoC-enabled environment